### ISSUE TITLE
Null filtering not working when using Grouping Sets

### ISSUE THREAD

**Issue by skokenes** [duckdb/duckdb#6686]
### What happens?

If I run a GROUPING SET query with a HAVING clause at the end to filter a column to non-null values, the null values are still included.
Query:
```sql
SELECT 
  i, 
  j, 
  count(*) as count 
FROM 
  (SELECT 
     i%3 as i, 
     i%4 as j 
    FROM range(100) t(i)
  ) 
GROUP BY GROUPING SETS(i,j) 
HAVING i IS NOT NULL;
```
Resulting table:
<img width="141" alt="image" src="https://user-images.githubusercontent.com/5554373/224509104-14567c42-9903-4284-97d2-2e4a30b21bc0.png">

My expectation is that the null rows for i would not be included in the result.
Oddly enough, this query DOES work as expected if I also add an empty grouping set `()`:
Query: 
```sql
SELECT 
  i, 
  j, 
  count(*) as count 
FROM 
  (SELECT 
     i%3 as i, 
     i%4 as j 
    FROM range(100) t(i)
  ) 
GROUP BY GROUPING SETS(i,j,()) 
HAVING i IS NOT NULL;
```
Resulting table:
<img width="140" alt="image" src="https://user-images.githubusercontent.com/5554373/224509166-39c14221-7dfc-4537-bb1e-6da427aaf7e2.png">

### To Reproduce

Run this query and see that the resulting `i` column includes null values:
```sql
SELECT 
  i, 
  j, 
  count(*) as count 
FROM 
  (SELECT 
     i%3 as i, 
     i%4 as j 
    FROM range(100) t(i)
  ) 
GROUP BY GROUPING SETS(i,j) 
HAVING i IS NOT NULL;
```

### OS:

OSX x64

### DuckDB Version:

@duckdb/duckdb-wasm@1.24.1-dev7.0

### DuckDB Client:

shell.duckdb.org

### Full Name:

Speros Kokenes

### Affiliation:

Qlik

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by paul-iqmo** [comment]
Since I didn't know what to expect, here's a comparison vs MS SQL Server:
```
1> WITH x AS (SELECT n FROM (VALUES (0),(1),(2),(3),(4),(5),(6),(7),(8),(9)) v(n))
2> select * from (select n%3 as i, n%4 as j from x) as k
3> group by grouping sets(i,j)
4> having j is not null
5> go
i           j
----------- -----------
       NULL           0
       NULL           1
       NULL           2
       NULL           3

(4 rows affected)
1> SELECT @@version;
2> go
                                                                                                                                                                                                                                                
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Microsoft SQL Server 2022 (RTM) - 16.0.1000.6 (X64)
        Oct  8 2022 05:58:25
        Copyright (C) 2022 Microsoft Corporation
        Express Edition (64-bit) on Windows 10 Pro 10.0 <X64> (Build 19045: ) (Hypervisor)


(1 rows affected)
```

**Comment by skokenes** [comment]
Worth noting the same problem also exists if you wrap with an outer SELECT statement using WHERE instead of HAVING to try and filter the nulls

**Comment by Mytherin** [comment]
Thanks for the report! This issue was actually recently found by @PedroTadim as part of his fuzzing efforts and I have pushed a fix for that in #6681. The issue was that filters were incorrectly being pushed down `GROUPING SETS` aggregates. The filters would be executed **before** the grouping set, which caused any null values introduced by the grouping set to still be included in the final result. The current master produces the correct result:

```sql
D SELECT 
>   i, 
>   j, 
>   count(*) as count 
> FROM 
>   (SELECT 
>      i%3 as i, 
>      i%4 as j 
>     FROM range(100) t(i)
>   ) 
> GROUP BY GROUPING SETS(i,j) 
> HAVING i IS NOT NULL;
┌───────┬───────┬───────┐
│   i   │   j   │ count │
│ int64 │ int64 │ int64 │
├───────┼───────┼───────┤
│     0 │  NULL │    34 │
│     1 │  NULL │    33 │
│     2 │  NULL │    33 │
└───────┴───────┴───────┘
```

**Comment by Mytherin** [comment]
Appears to be fixed in master
